RevenueCatUI/View+PresentPaywall.swift

```diff
@@ -65,18 +67,77 @@ extension View {
     /// [Documentation](https://rev.cat/paywalls)
     ///
     /// - Tag: presentPaywallIfNeeded
+    @available(iOS, deprecated: 1, renamed: "presentPaywallIfNeeded(requiredEntitlementIdentifier:offering:fonts:presentationMode:purchaseStarted:purchaseCompleted:purchaseCancelled:restoreStarted:restoreCompleted:purchaseFailure:restoreFailure:onDismiss:)")
+    @available(tvOS, deprecated: 1, renamed: "presentPaywallIfNeeded(requiredEntitlementIdentifier:offering:fonts:presentationMode:purchaseStarted:purchaseCompleted:purchaseCancelled:restoreStarted:restoreCompleted:purchaseFailure:restoreFailure:onDismiss:)")
+    @available(watchOS, deprecated: 1, renamed: "presentPaywallIfNeeded(requiredEntitlementIdentifier:offering:fonts:presentationMode:purchaseStarted:purchaseCompleted:purchaseCancelled:restoreStarted:restoreCompleted:purchaseFailure:restoreFailure:onDismiss:)")
+    @available(macOS, deprecated: 1, renamed: "presentPaywallIfNeeded(requiredEntitlementIdentifier:offering:fonts:presentationMode:purchaseStarted:purchaseCompleted:purchaseCancelled:restoreStarted:restoreCompleted:purchaseFailure:restoreFailure:onDismiss:)")
+    @available(macCatalyst, deprecated: 1, renamed: "presentPaywallIfNeeded(requiredEntitlementIdentifier:offering:fonts:presentationMode:purchaseStarted:purchaseCompleted:purchaseCancelled:restoreStarted:restoreCompleted:purchaseFailure:restoreFailure:onDismiss:)")
+    // swiftlint:enable line_length
     public func presentPaywallIfNeeded(
         requiredEntitlementIdentifier: String,
         offering: Offering? = nil,
         fonts: PaywallFontProvider = DefaultPaywallFontProvider(),
         presentationMode: PaywallPresentationMode = .default,
-        purchaseStarted: PurchaseStartedHandler? = nil,
+        purchaseStarted: @escaping PurchaseStartedHandler,
```

Comment:
Made this non nullable so the function wouldn't collide if someone was only passing `requiredEntitlementIdentifier`. I also had to made it `@escaping`. Is that considered a breaking change?

Reply:
@vegaro I'm not certain if the `@escaping` is breaking changing (I'd have to experiment) but making this non-nullable is a breaking change, isn't it? 😬 

If somebody was using this function without `purchaseStarted`, it will now require them to have `purchaseStarted`?

Reply:
Did the API tester for this break? That would really let us know, wouldn't it? 🤷‍♂️

Reply:
It didn't fail, that's why I am not sure, but I think it's not

> If somebody was using this function without purchaseStarted, it will now require them to have purchaseStarted?
If they were not passing it, it will use the new non-deprecated function automatically (that one has it as nullable)

Reply:
Ahhh, okay! That makes sense then 😊 I think this should be good then!